src/utils/react-query-factory.js

```diff
@@ -0,0 +1,72 @@
+// The keys that map to enterpriseQueryKeys that is used to define specific queries
+// and used to specifically invalidate queries
+export const queryKeys = {
+  // Global key to all queries
+  all: ['enterprise'],
+  enterprise: (enterpriseUuid) => [...queryKeys.all, enterpriseUuid],
+  entitlements: ['user', 'entitlements'],
+  enterpriseLearner: () => [...queryKeys.all, 'linked-enterprise-customer-users'],
+  enterpriseCourseEnrollments: (enterpriseUuid) => [...queryKeys.enterprise(enterpriseUuid), 'enrollments'],
+  enterpriseCourseMetadata: (enterpriseUuid) => [...queryKeys.enterprise(enterpriseUuid), 'course'],
+  canRedeem: (enterpriseUuid) => [...queryKeys.enterpriseCourseMetadata(enterpriseUuid), 'can-redeem'],
+  // By feature prepended by enterpriseUuid
+  enterpriseCuration: (enterpriseUuid) => [...queryKeys.enterprise(enterpriseUuid), 'content-highlights'],
```

Comment:
[suggest] Perhaps named `contentHighlights`? Can likely keep the `enterpriseCuration` as a key specific to the `enterprise-curations` API call itself.